components/eventing-controller/upgrade-job/clients/deployment/deployment.go

```diff
@@ -0,0 +1,69 @@
+package deployment
+
+import (
+	"context"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+type Client struct {
+	dynamicClient dynamic.Interface
+}
+
+func NewClient(client dynamic.Interface) Client {
+	return Client{dynamicClient: client}
+}
+
+func (c Client) Get(namespace, name string) (*appsv1.Deployment, error) {
+	unstructuredDeployment, err := c.dynamicClient.Resource(GroupVersionResource()).Namespace(namespace).Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
Is the context unclear? Otherwise take `context.Background()`. See [godoc](https://pkg.go.dev/context@go1.16.6#TODO).